File: ArendInventar/src/main/java/com/arendinventar/model/Equipment.java
package com.arendinventar.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Equipment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idEquipment;

    @Column
    private String nameEquip;

    private String description;

    @Column
    private double costArend;

    @Column
    private double deposit;

    @ManyToOne
    @JoinColumn(name = "idTypeEquipment")
    private TypeEquipment typeEquipment;

    @ManyToOne
    @JoinColumn(name = "idViewEquipment")
    private ViewEquipment viewEquipment;

    @Column(nullable = false)
    private String numberEquipment;

}

File: WebArend/src/app/model/equipment.ts
export interface Equipment {
  id: number;
  nameEquip: string;
  description: string;
  costArend: number;
  deposit: number;
  typeEquipment: TypeEquipment;
  viewEquipment: ViewEquipment;
  numberEquipment: number;
  imageUrl: string;
  quantity: number;
}

export interface TypeEquipment {
  id: number;
  nameTypeEquipment: string;
}

export interface ViewEquipment {
  id: number;
  nameViewEquipment: string;
  cost: number;
}
